testing/adios2/transform/TestBZip2Wrapper.cpp

```diff
@@ -0,0 +1,125 @@
+#include <cstdint>
+#include <iostream>
+#include <numeric> //std::iota
+#include <stdexcept>
+
+#include <adios2.h>
+
+#include <gtest/gtest.h>
+
```

Comment:
This is a good start for tests.  It would be even better to add 2 more, one that compresses natively with bzip2 api and decompresses with adios wrapper, then another that compresses with adios wrapper and decompresses with native bzip2 api.